src/server/middleware/auth.ts

```diff
@@ -449,3 +482,28 @@ const hashRequestBody = (req: FastifyRequest): string => {
     .update(JSON.stringify(req.body), "utf8")
     .digest("hex");
 };
+
+/**
+ * Check if the request IP is in the allowlist.
+ * Fetches cached config if available.
+ * env.TRUST_PROXY is used to determine if the X-Forwarded-For header should be trusted.
+ * @param req FastifyRequest
+ * @returns boolean
+ * @async
+ */
+const checkIpInAllowlist = async (req: FastifyRequest) => {
+  const config = await getConfig();
+
+  // can be simplified to:
+  // !(config.ipAllowlist.length > 0 && !config.ipAllowlist.includes(req.ip))
+  // but it's more readable this way
+
+  if (config.ipAllowlist.length > 0) {
+    if (config.ipAllowlist.includes(req.ip)) {
+      return true;
+    }
+    return false;
+  }
+
+  return true;
```

Comment:
Nit: it's more readable to me to handle the common/base case first (no IP allowlist):

```suggestion
if (config.ipAllowlist.length === 0) {
  return true;
}
return config.ipAllowlist.includes(req.ip);
```